### ISSUE TITLE
git.get_git_dir() appears to break with older versions of git

### ISSUE THREAD

**Issue by mmacpherson** [pre-commit/pre-commit#831]
When you install the current pre-commit (v1.11.0) on a system using an older git, eg 2.1.4, the `git rev-parse --git-common-dir` ultimately issued by `git.get_git_dir()` yields the result `--git-common-dir` instead of `.git` like you'd expect. This causes pre-commit to get installed in a directory literally named "--git-common-dir", and thus not work.

Happened to encounter using a docker image for an older version of python, which used an older git. Examples to demo the issue:

```
$ docker run python:2.7.9 sh -c "git init foo && cd foo && git rev-parse --git-common-dir && git --version"
Initialized empty Git repository in /foo/.git/
--git-common-dir
git version 2.1.4
$ docker run python:2.7.15 sh -c "git init foo && cd foo && git rev-parse --git-common-dir && git --version"
Initialized empty Git repository in /foo/.git/
.git
git version 2.11.0
```

In git.get_git_dir() as it stands, it appears that the older-git behavior is anticipated. It looks like older gits are expected to throw a `CalledProcessError` when `git rev-parse --git-common-dir` is called, and fall back to `git rev-parse --git-dir`. But evidently that's not working as intended?

**Comment by asottile** [comment]
yikes!  Thanks for the report I'll get a fix for this

Looks like this was depending on the PARSEOPT mode of `rev-parse` which isn't the default

**Comment by asottile** [comment]
This is a regression introduced in [v1.10.5](https://github.com/pre-commit/pre-commit/releases/tag/v1.10.5)

**Comment by asottile** [comment]
actually hmm, parseopt seems insufficient as well

**Comment by asottile** [comment]
#832 has a fix for this, turns out it's difficult to make `rev-parse` error on unknown options.  When I originally tried this I must have been outside a git directory or something :S

**Comment by mmacpherson** [comment]
Very cool, thanks much for looking into it so quickly, and for this very useful package.

**Comment by asottile** [comment]
thanks for the report! I'll close this once I've merged the fix -- looks like I need to attend to some tests 😆

**Comment by asottile** [comment]
This has been released as part of [v1.11.1](https://github.com/pre-commit/pre-commit/releases/tag/v1.11.1) -- thanks again for the issue 🎉
